flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/catalog/ExternalCatalogTable2TableSourceConverter.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.plan.catalog
+
+import org.apache.flink.table.sources.TableSource
+
+/** Defines a converter used to convert [[org.apache.flink.table.sources.TableSource]] to or from [[ExternalCatalogTable]].
+  *
+  * @tparam T The tableSource type which to do convert operation on.
+  */
+trait ExternalCatalogTable2TableSourceConverter[T <: TableSource[_]] {
```

Comment:
Please rename. Do not use number "2" in a trait. Please use noun or adjectives. TableSourceConverter may be a good idea.